src/sendqueue.rs

```diff
@@ -0,0 +1,93 @@
+use std::convert::TryInto;
+use std::ptr::NonNull;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use libc::{c_int, c_uint};
+
+use crate::raw;
+use crate::Error;
+use crate::{Active, Capture};
+
+pub struct SendQueue {
+    squeue: NonNull<raw::pcap_send_queue>,
+    sync: c_int,
+}
+
+impl SendQueue {
+    pub fn new(memsize: c_uint) -> Result<Self, Error> {
+        let squeue = unsafe { raw::pcap_sendqueue_alloc(memsize) };
+        let squeue = if let Some(squeue) = NonNull::new(squeue) {
+            squeue
+        } else {
+            return Err(Error::InsufficientMemory);
+        };
+
+        Ok(Self { squeue, sync: 0 })
+    }
+
+    pub fn maxlen(&self) -> c_int {
+        unsafe { (*self.squeue.as_ptr()).maxlen }
+    }
+
+    pub fn len(&self) -> c_int {
+        unsafe { (*self.squeue.as_ptr()).len }
+    }
+
+    pub fn sync(&mut self, sync: bool) {
+        self.sync = if sync { 1 } else { 0 }
```

Comment:
`bool` in Rust are guarantee to value 1 for true and 0 for false https://doc.rust-lang.org/std/primitive.bool.html